joint_names_arr: &jointNames
    - panda_joint1
    - panda_joint2
    - panda_joint3
    - panda_joint4
    - panda_joint5
    - panda_joint6
    - panda_joint7

cartesian_impedance_controller_softbots:
    type: panda_controllers/CartesianImpedanceControllerSoftbots
    arm_id: panda
    joint_names: *jointNames

computed_torque_controller:
    type: panda_controllers/ComputedTorque
    arm_id: panda
    joint_names: *jointNames
    cutoff_frequency: 1000
    # --- Controller parameters --- #
    kp1: 30.0   #default: 120.0
    kp2: 15.0   #default: 70.0
    kp3: 3.0    #default: 50.0
    kv1: 30.0
    kv2: 15.0
    kv3: 3.0
    dt: 0.001
    # gainRlinks: [0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01]                   # gain for R matrix
    gainRlinks: [1, 1, 1, 1, 1, 1, 1] 
    gainRparam: [1, 0.1, 0.01]                          # gain proportional to [mass, mass*length, mass*length^2] 
    update_param: false
    # P11: [0.5349, 0.0667, 0.0667, 0.0667, 0.0836, 0.0836, 0.2222]
    # P22: [0.4703, 5.0167, 5.0167, 5.0167, 5.0333, 5.0333, 5.1667]
    # Pout: [-0.5, -0.5, -0.5, -0.5, -0.5, -0.5, -0.5]
    P11: [0.0052, 0.0052, 0.0052, 0.0052, 0.0054, 0.0054, 0.0072]
    P22: [0.0052, 0.0052, 0.0052, 0.0052, 0.0053, 0.0053, 0.0067]
    Pout: [-0.005, -0.005, -0.005, -0.005, -0.005, -0.005, -0.005]

computed_torque_mod_controller:
    type: panda_controllers/ComputedTorqueMod
    arm_id: panda
    joint_names: *jointNames
    cutoff_frequency: 1000
    # --- Controller parameters --- #
    kp1: 30.0   #default: 120.0
    kp2: 15.0   #default: 70.0
    kp3: 3.0    #default: 50.0
    kv1: 15.0
    kv2: 7.5
    kv3: 1.5
    dt: 0.001
    gainRlinks: [1, 1, 1, 1, 1, 1, 1]                   # gain for R matrix
    gainRparam: [1, 1, 1, 1]
    # gainRparam: [1, 0.1, 0.01, 0.001]                          # gain proportional to [mass, mass*length, mass*length^2] 
    update_param: false

CT_mod_controller_OS:
    type: panda_controllers/CTModOS
    arm_id: panda
    joint_names: *jointNames
    cutoff_frequency: 1000
    # --- Controller parameters --- #
    # stiffness terms
    kp1: 50.0   #default: 50
    kp2: 50.0
    kp3: 50.0
    kp4: 1.0
    kp5: 1.0
    kp6: 1.0
    # damping terms
    kv1: 20.0   #default: 20
    kv2: 20.0
    kv3: 20.0
    kv4: 0.5
    kv5: 0.5
    kv6: 0.5
    # joints level terms, null control
    kv1: 15.0
    kv2: 5.0
    kv3: 2.0
    kn1: 5.0
    kn2: 3.0
    kn3: 1.0
    dt: 0.001
    # gainRlinks: [0, 0, 0, 0, 0, 0, 1]                   # gain for R matrix
    gainRlinks: [1, 1, 1, 1, 1, 1, 1]                   # gain for R matrix
    # gainRparam: [1, 0.1, 0.01, 0.001]                   # gain proportional to [mass, mass*length, mass*length^2]
    gainRparam: [1, 1, 1, 1] 
    gainLambda: [2.0, 2.0, 3.0, 5.0, 5.0, 5.0] 
    update_param: false
   

backstepping_controller:
    type: panda_controllers/Backstepping
    arm_id: panda
    joint_names: *jointNames
    cutoff_frequency: 1000
    # --- Controller parameters --- #
    gainLambda: [5.0, 5.0, 5.0, 2.0, 2.0, 2.0]       # gain for Lambda matrix
    gainRlinks: [1, 1, 1, 1, 1, 1, 1]     # gain for R matrix
    gainRparam: [1.0, 0.1, 0.1, 0.1]                          # gain proportional to [mass, mass*length, mass*length^2] 
    gainKd: [10, 10, 10, 10, 5, 5, 3]                   # gain for Kd matrix
    update_param: false                                 # bool to update inertial parameters
    upper_bound_s: true
    
